hummingbot/connector/exchange/btc_markets/btc_markets_utils.py

```diff
@@ -0,0 +1,98 @@
+import math
+from typing import Dict, List
+from dateutil.parser import parse as dateparse
+
+from hummingbot.core.utils.tracking_nonce import get_tracking_nonce, get_tracking_nonce_low_res
+from . import btc_markets_constants as Constants
+
+from hummingbot.client.config.config_var import ConfigVar
+from hummingbot.client.config.config_methods import using_exchange
+
+
+CENTRALIZED = True
+
+EXAMPLE_PAIR = "XRP-AUD"
+
+# Using AUD Market Pair rates, but need to figure out how to determine 30 day AUD vol for rolling
+# https://www.btcmarkets.net/fees
+DEFAULT_FEES = [0.85, 0.85]
+
+HBOT_BROKER_ID = "HBOT-"
+
+
+# deeply merge two dictionaries
+def merge_dicts(source: Dict, destination: Dict) -> Dict:
+    for key, value in source.items():
+        if isinstance(value, dict):
+            # get node or create one
+            node = destination.setdefault(key, {})
+            merge_dicts(value, node)
+        else:
+            destination[key] = value
+
+    return destination
+
+
+# join paths
+def join_paths(*paths: List[str]) -> str:
+    return "/".join(paths)
+
+
+# get timestamp in milliseconds
+def get_ms_timestamp() -> int:
+    return get_tracking_nonce_low_res()
+
+
+# convert milliseconds timestamp to seconds
+def ms_timestamp_to_s(ms: int) -> int:
+    return math.floor(ms / 1e3)
+
+
+# convert date string to timestamp
+def str_date_to_ts(date: str) -> int:
+    return int(dateparse(date).timestamp())
+
+
+# Request ID class
+class RequestId:
+    """
+    Generate request ids
+    """
+    _request_id: int = 0
+
+    @classmethod
+    def generate_request_id(cls) -> int:
+        return get_tracking_nonce()
+
+
+def convert_from_exchange_trading_pair(exchange_trading_pair: str) -> str:
+    return exchange_trading_pair.replace("_", "-")
+
+
+def convert_to_exchange_trading_pair(hb_trading_pair: str) -> str:
+    return hb_trading_pair.replace("-", "_")
+
+
+def get_new_client_order_id(is_buy: bool, trading_pair: str) -> str:
+    side = "B" if is_buy else "S"
+    return f"{HBOT_BROKER_ID}{side}-{trading_pair}-{get_tracking_nonce()}"
```

Comment:
Please double check it is really required to implement a new function to generate the order id instead of using the generic `get_new_client_order_id` function in `hummingbot.connector.utils`

Reply:
Based on crypto.com connector - https://github.com/hummingbot/hummingbot/blob/master/hummingbot/connector/exchange/crypto_com/crypto_com_utils.py#L68